elifepubmed/generate.py

```diff
@@ -601,6 +601,45 @@ def set_grants(parent, poa_article):
                 set_object(parent, "grant", params)
 
 
+def dataset_assigning_authority(assigning_authority, uri):
+    """precise assigning_authority value considering the uri in some cases"""
+    authority_map = OrderedDict([('NCBI', [])])
```

Comment:
This could probably be a constant. I personally would advice against `OrderedDict` unless strictly necessary (which it is usually not, but we had discussions before).

Reply:
I converted it to a constant, but I can probably get rid of the `OrderedDict` usage. I've often been defaulting to `OrderedDict` because of the inconsistent test case results we got with `dict`, but I think that was a very specific situation for when it was converted to JSON and then compared, the order mattered.

Reply:
Yes, true, if you compare the output of something iterating over a dict or other unordered iterables then the test could fail. But in those cases it would maybe have tested an assumption regarding the implementation, rather than what the "contract" of the function probably should have been. (apparently [JSON objects are unordered](http://www.json.org/) - `An object is an unordered set of name/value pairs`)

Maybe in that case it could just parse the JSON again to dict and compare that?

In any case it looks like [dict becomes more ordered](https://mail.python.org/pipermail/python-dev/2017-December/151283.html).